core/src/main/java/org/elasticsearch/rest/action/search/RestSearchScrollAction.java

```diff
@@ -58,25 +57,33 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
             searchScrollRequest.scroll(new Scroll(parseTimeValue(scroll, null, "scroll")));
         }
 
-        BytesReference body = request.contentOrSourceParam();
-        if (body.length() > 0) {
-            if (XContentFactory.xContentType(body) == null) {
-                if (scrollId == null) {
-                    scrollId = body.utf8ToString();
-                    searchScrollRequest.scrollId(scrollId);
+        request.withContentOrSourceParamParserOrNull(xContentParser -> {
+            if (xContentParser == null) {
+                if (request.hasContent()) {
+                    // TODO: why do we accept this plain text value? maybe we can just use the scroll params?
+                    BytesReference body = request.getContentOrSourceParamOnly();
```

Comment:
in this plain/text case does one have to provide the header? not yet in 5.x but they will need to in master?